pubsub/envelope.go

```diff
@@ -121,6 +122,44 @@ func NewCloudEventsEnvelope(id, source, eventType, subject string, topic string,
 	return ce
 }
 
+// NewCloudEventsHeaders returns a map representation of a cloudevents headers.
+func NewCloudEventsHeaders(id, source, eventType, subject string, topic string, pubsubName string,
+	dataContentType string, traceParent string, traceState string,
+) map[string]interface{} {
+	// defaults
+	if id == "" {
+		id = uuid.New().String()
+	}
+	if source == "" {
+		source = DefaultCloudEventSource
+	}
+	if eventType == "" {
+		eventType = DefaultCloudEventType
+	}
+	if dataContentType == "" {
+		dataContentType = DefaultCloudEventDataContentType
+	}
+
+	ce := map[string]interface{}{
+		"ce-" + IDField:              id,
```

Comment:
I think these should all be `ce_` according to the [kafka protocol binding spec](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/bindings/kafka-protocol-binding.md#3231-property-names)

Reply:
Alright, I updated; however, I think this PR needs some work/adjustment. The pubsub implementation will need to manage properly serializing/deserializing

Each binding has a different way to place the data:

- http: `ce-` prefixed headers 
- amqp: `cloudEvents:` prefixed properties
- kafka: `ce_` prefixed headers

Reply:
Yikes, I didn't realize there were different prefixes for different things...